File: Go/Stack.go
package main
import "fmt"

type Node struct {
	data int
	link *Node
}

var head *Node

func push(data int){
	if head == nil {
		head = &Node{data, (&Node{})}
	} else {
		temp := &Node{data, head}
		head = temp
	}
}

func pop() int {
	if (head == nil){
		fmt.Println("Stack Empty!")
	}
	data := head.data
	head = head.link
	return data
}

func peek() int {
	if head == nil {
		fmt.Println("Stack Empty!")
	}
	return head.data
}

func isEmpty() bool {
	if (head == nil){
		return true
	}
	return false
}

func size() int {
	temp := head
	var counter int = 0
	for {
		if temp == nil {
			break
		}
		counter++
		temp = temp.link
	}
	return counter
}

func printList(){
	temp := head
	for {
		if temp == nil {
			break
		}
		fmt.Println(temp.data)
		temp = temp.link
	}
}

func main(){
	fmt.Println(isEmpty())

	push(1)
	push(3)
	push(5)
	push(7)
	push(9)

	printList()

	fmt.Println(isEmpty())
	fmt.Println(size())
	fmt.Println(peek())
	fmt.Println(pop())

	printList()
}

File: C/Stack.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node *link;
};

struct Node *head = NULL;

void push(int data);
int pop();
int peek();
bool isEmpty();
int size();
void printList();

int main(){
	printf("%d", isEmpty());
	push(1);
	push(3);
	push(5);
	push(7);
	push(9);

	printList();
	printf("%d\n", size());
	printf("%d\n", isEmpty());
	printf("%d\n", pop());
	printf("%d\n", isEmpty());

	printList();

    return 0;
}

void push(int data){
	struct Node *temp = malloc(sizeof(struct Node));
	if (head == NULL){
		temp->data = data;
		temp->link = NULL;
		head = temp;
	}
	else {
		temp->data = data;
		temp->link = head;
		head = temp;
	}
}

int peek (){
    struct Node *temp = head;
    if (temp != NULL){
        return temp->data;
    } else {
		printf("Stack Empty!");
	}
}

int pop(){
    if (head == NULL){
        printf("Stack Empty!");
    }
	int data = head->data;
    head = head->link;
	return data; 
}

bool isEmpty(){
	struct Node *temp = head;
	if (temp == NULL){
        return true;
    }
    return false;
}

int size(){
	struct Node *temp = head;
	int counter = 0;
	while(temp != NULL){
		temp = temp->link;
		counter++;
	}
	return counter;
}

void printList(){
	struct Node *temp = head;
	while(temp != NULL){
		printf("%d\n", temp->data);
		temp = temp->link;
	}
}
